utils/wasm-gen/src/config/syscalls/param.rs

```diff
@@ -172,3 +234,40 @@ impl SyscallParamAllowedValues {
         unstructured.int_in_range(self.0.clone())
     }
 }
+
+impl From<RangeInclusive<i64>> for RegularParamAllowedValues {
+    fn from(range: RangeInclusive<i64>) -> Self {
+        Self(range)
+    }
+}
+
+impl Default for RegularParamAllowedValues {
+    fn default() -> Self {
+        Self::zero()
+    }
+}
+
+/// Allowed values for syscalls pointer params.
+///
+/// Currently it allows defining only actor kinds (`SyscallDestination`)
+/// and message values for syscalls that send messages to actors.
+// TODO #3591 Support other hash types.
+#[derive(Debug, Clone)]
+pub enum PtrParamAllowedValues {
+    Value(RangeInclusive<u128>),
+    ActorIdWithValue {
+        actor: SyscallDestination,
```

Comment:
`SyscallDestination` is not an `actor`, maybe rename it to something else?

Reply:
renaming `SyscallDestination` to `ActorKind`.

Reply:
done here
[2668531](https://github.com/gear-tech/gear/pull/3625/commits/266853145bfc59843fb3ad8b9407d52e1a9559b4)